scripts/validate_docstrings.py

```diff
@@ -331,6 +336,41 @@ def parameter_mismatches(self):
 
         return errs
 
+    @property
+    def pep8_violations(self):
+        with self._file_representation() as file:
+            application = Flake8()
+            application.initialize(["--doctests", "--quiet"])
+            application.run_checks([file.name])
+            application.report()
+            stats = application.guide.stats
+            return [
+                "{} {} {}".format(s.count, s.error_code, s.message)
+                for s in stats.statistics_for('')
+            ]
+
+    @contextmanager
+    def _file_representation(self):
+        """
+        Temporarily creates file with current function inside.
+        The signature and body are **not** included.
+
+        :returns file
+        """
+        create_function = 'def {name}():\n' \
+                          '    """{doc}"""\n' \
+                          '    pass\n'
+
+        name = self.name.split('.')[-1]
+        lines = self.clean_doc.split("\n")
+        indented_lines = [(' ' * 4) + line if line else ''
+                          for line in lines[1:]]
+        doc = '\n'.join([lines[0], *indented_lines])
```

Comment:
I'm a bit confused with this. I guess I'm missing something. If in our docstring we have something like:
```
Some desc.

Examples
--------
>>> val = 1+2
>>> print(val)
```

I'd expect to have in the temporary file to validate:

```
val = 1+2
print(val)
```

As we won't validate the description, or anything else. I think numpydoc returns the lines with code, so writing to the file should be a single line of code. Is there any advantage generating a function?

Reply:
Well flake8 supports doctests by adding `--doctests` as an argument. And the Examples follow the doctest standard.

> Examples
> An optional section for examples, using the doctest format. 

Having to carve out the examples is not trivial and would require writing multiple tests. So creating a file with a dummy-function is a simple hack, getting ugly cause it has to restore the indentation of the docstring. One test of the `BadExamples` class ensures it is working.